presto-main/src/main/java/com/facebook/presto/sql/Serialization.java

```diff
@@ -107,8 +108,13 @@ public void serialize(VariableReferenceExpression value, JsonGenerator jsonGener
     public static class VariableReferenceExpressionDeserializer
             extends KeyDeserializer
     {
-        private final TypeManager typeManager;
+        private TypeManager typeManager = createTestFunctionAndTypeManager();
```

Comment:
Let's not make typeManager to be a "Test" one. Was there any reason you initialize it this way?

Reply:
this is used by test code path when jackson binder uses a default constructor that has no arg, I add a comment to clarify around the code block

Reply:
Unit tests from testcase `TestExplainVerification` would run into this code path when `TestingPrestoServer` boostraps `JsonCodecProvider`

Reply:
@yingsu00 thanks for suggestion. I have refactored the code to remove the TestingFunctionAndTypeManager from mainline logic and add a mockup deserializer for testing purpose to enable test coverage.